packages/manager/src/features/IAM/hooks/useIsIAMEnabled.test.ts

```diff
@@ -0,0 +1,116 @@
+import { renderHook, waitFor } from '@testing-library/react';
+
+import { accountPermissionsFactory } from 'src/factories/accountPermissions';
+import { http, HttpResponse, server } from 'src/mocks/testServer';
+import { wrapWithTheme } from 'src/utilities/testHelpers';
+
+import { useIsIAMEnabled } from './useIsIAMEnabled';
+
+const queryMocks = vi.hoisted(() => ({
+  useAccountPermissions: vi.fn().mockReturnValue({ foo: 'bar' }),
+}));
+
+vi.mock(import('src/queries/iam/iam'), async (importOriginal) => {
+  const actual = await importOriginal();
+  return {
+    ...actual,
+    useAccountPermissions: queryMocks.useAccountPermissions,
+  };
+});
+
+describe('useIsIAMEnabled', () => {
+  it('should correctly for IAM beta, enabled, API access', async () => {
+    const rolePermissions = accountPermissionsFactory.build();
+    server.use(
+      http.get('*/v4beta/iam/role-permissions', () => {
+        return HttpResponse.json(rolePermissions);
+      })
+    );
+
+    queryMocks.useAccountPermissions.mockReturnValue({
+      data: rolePermissions,
+    });
+
+    const flags = { iam: { beta: true, enabled: true } };
+
+    const { result } = renderHook(() => useIsIAMEnabled(), {
+      wrapper: (ui) => wrapWithTheme(ui, { flags }),
+    });
+
+    await waitFor(() => {
+      expect(result.current.isIAMBeta).toBe(true);
+      expect(result.current.isIAMEnabled).toBe(true);
+    });
+  });
+
+  it('should correctly for IAM port-beta, enabled, API access', async () => {
```

Comment:
```suggestion
  it('should correctly for IAM post-beta, enabled, API access', async () => {
```
nit: can we improve these test names? The suggestion above fixes a typo, but none of these tests are named very clearly imo

Reply:
I've updated the names, let me know if you had something else in mind.